e2e/test/scenarios/dashboard-filters/reproductions/16663-filters-can-stay-in-url.cy.spec.js

```diff
@@ -0,0 +1,64 @@
+import {
+  restore,
+  popover,
+  editDashboard,
+  saveDashboard,
+  visitDashboard,
+} from "e2e/support/helpers";
+
+import { SAMPLE_DATABASE } from "e2e/support/cypress_sample_database";
+
+const { ORDERS_ID } = SAMPLE_DATABASE;
+
+const questionDetails = {
+  query: {
+    "source-table": ORDERS_ID,
+  },
+};
+
+const parameters = [
+  {
+    name: "Quarter and Year",
+    slug: "quarter_and_year",
+    id: "f8ae0c97",
+    type: "date/quarter-year",
+    sectionId: "date",
+  },
+];
+
+const dashboardDetails = { parameters };
+
+describe("issue 16663", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("should remove filter value from url after going to another dashboard (metabase#16663)", () => {
+    cy.createQuestionAndDashboard({ questionDetails, dashboardDetails }).then(
+      ({ body: dashboardCard }) => {
+        const { dashboard_id } = dashboardCard;
+
+        visitDashboard(dashboard_id);
+      },
+    );
+
+    editDashboard();
+
+    cy.get("main header").find(".Icon-gear").click();
```

Comment:
could you try if 
```suggestion
    dashboardHeader().icon("gear").click();
```
would work? I think using selectors makes the test hard to follow.

Reply:
I updated code to use `icon`, but dashboardHeader is 3 levels down comparing to what I need, so it didn't work